mdata/notifierKafka/notifierKafka.go

```diff
@@ -4,202 +4,85 @@ import (
 	"bytes"
 	"encoding/binary"
 	"encoding/json"
+	"hash/fnv"
 	"sync"
 	"time"
 
-	schema "gopkg.in/raintank/schema.v1"
-
-	"github.com/Shopify/sarama"
+	confluent "github.com/confluentinc/confluent-kafka-go/kafka"
+	part "github.com/grafana/metrictank/cluster/partitioner"
 	"github.com/grafana/metrictank/idx"
 	"github.com/grafana/metrictank/kafka"
 	"github.com/grafana/metrictank/mdata"
 	"github.com/grafana/metrictank/util"
 	"github.com/raintank/worldping-api/pkg/log"
+	schema "gopkg.in/raintank/schema.v1"
 )
 
 type NotifierKafka struct {
-	instance  string
-	in        chan mdata.SavedChunk
-	buf       []mdata.SavedChunk
-	wg        sync.WaitGroup
-	idx       idx.MetricIndex
-	metrics   mdata.Metrics
-	bPool     *util.BufferPool
-	client    sarama.Client
-	consumer  sarama.Consumer
-	producer  sarama.SyncProducer
-	offsetMgr *kafka.OffsetMgr
-	StopChan  chan int
-
-	// signal to PartitionConsumers to shutdown
-	stopConsuming chan struct{}
+	instance    string
+	in          chan mdata.SavedChunk
+	buf         []mdata.SavedChunk
+	wg          sync.WaitGroup
+	bPool       *util.BufferPool
+	idx         idx.MetricIndex
+	metrics     mdata.Metrics
+	partitioner *part.Kafka
+	consumer    *kafka.Consumer
+	producer    *confluent.Producer
+	stopChan    chan struct{}
 }
 
 func New(instance string, metrics mdata.Metrics, idx idx.MetricIndex) *NotifierKafka {
-	client, err := sarama.NewClient(brokers, config)
-	if err != nil {
-		log.Fatal(2, "kafka-cluster failed to start client: %s", err)
-	}
-	consumer, err := sarama.NewConsumerFromClient(client)
-	if err != nil {
-		log.Fatal(2, "kafka-cluster failed to initialize consumer: %s", err)
-	}
-	log.Info("kafka-cluster consumer initialized without error")
+	producer, err := confluent.NewProducer(kafka.GetConfig(consumerConf.Broker, "snappy", consumerConf.BatchNumMessages, consumerConf.BufferMaxMs, consumerConf.ChannelBufferSize, consumerConf.FetchMin, consumerConf.NetMaxOpenRequests, consumerConf.MaxWaitMs, consumerConf.SessionTimeout))
```

Comment:
it's confusing that we're creating a producer using a consumerconf. this can use some explanation

Reply:
right, i guess i should just rename `consumerConf` to `clientConf`